metricflow-semantics/metricflow_semantics/model/semantics/linkable_spec_resolver.py

```diff
@@ -244,25 +217,34 @@ def _get_semantic_model_for_measure(self, measure_reference: MeasureReference) -
             )
         return semantic_models_where_measure_was_found[0]
 
-    def _get_joinable_metrics_for_semantic_model(self, semantic_model: SemanticModel) -> LinkableElementSet:
-        linkable_metrics = []
-        for metric_ref in self._joinable_metrics_for_semantic_models.get(semantic_model.reference, set()):
-            for entity_link in [entity.reference for entity in semantic_model.entities]:
-                linkable_metrics.append(
-                    LinkableMetric(
-                        properties=frozenset({LinkableElementProperty.METRIC, LinkableElementProperty.JOINED}),
-                        join_path=MetricSubqueryJoinPath(
-                            metric_subquery_join_path_element=MetricSubqueryJoinPathElement(
-                                metric_reference=metric_ref, join_on_entity=entity_link
-                            )
+    def get_joinable_metrics_for_semantic_model(
+        self, semantic_model: SemanticModel, using_join_path: Optional[SemanticModelJoinPath] = None
+    ) -> LinkableElementSet:
+        """Get the set of linkable metrics that can be joined to this semantic model."""
+        properties = frozenset({LinkableElementProperty.METRIC, LinkableElementProperty.JOINED})
+        if using_join_path:
+            assert (
+                semantic_model.reference == using_join_path.last_semantic_model_reference
+            ), "Last join path element should match semantic model when building LinkableMetrics."
+            properties = properties.union(frozenset({LinkableElementProperty.MULTI_HOP}))
+
+        path_key_to_linkable_metrics: Dict[ElementPathKey, Tuple[LinkableMetric, ...]] = {}
+        for entity_reference in [entity.reference for entity in semantic_model.entities]:
+            if using_join_path and entity_reference in using_join_path.entity_links:
```

Comment:
I'm still not clear on why this is here. Is this to avoid entity link cycles?

I put this in the bigger PR but maybe a local variable to explain the purpose would help clear thing sup

Reply:
Correct, avoiding a cycle!
Oops I should go back and make sure I addressed all your previous comments...